packages/xo-web/src/xo-app/home/index.js

```diff
@@ -287,9 +287,17 @@ const TYPES = {
 
 const DEFAULT_TYPE = 'VM'
 
-@addSubscriptions({
+const NoObjects = props => {
+  if (props.isAdmin) {
+    return <NoObjectsWithServers {...props} />
+  }
+  return <NoObjectsWithoutServers {...props} />
+}
+
+const NoObjectsWithServers = addSubscriptions(props => ({
   noRegisteredServers: cb => subscribeServers(data => cb(isEmpty(data))),
-})
+}))(props => <NoObjectsWithoutServers {...props} />)
```

Comment:
`addSubscriptions` is a component decorator and this function is simply a component which wraps `NoObjectsWithoutServers` without doing anything.

You can decorate `NoObjectsWithoutServers` directly without wrapping it.